### PR TITLE
Make the folding environment working with ProvableTrace

### PR DESCRIPTION
**Pull request by dannywillems**
Allow to define a folding scheme without selectors, i.e. with only the columns that are really used. For MIPS, it is the 93 columsn for the state + the error term and the halt.

The pull request includes functions to build a whole trace for a single MIPS instruction, addiu, and can be replicated for other.

### CODE REVIEW

File: o1vm/src/mips/tests.rs [hunk 1]
```diff
@@ -323,4 +335,200 @@ mod folding {
...
+            build_folding_instance(&folding_witness_one, &mut fq_sponge, domain, &srs);
+        let folding_instance_two =
+            build_folding_instance(&folding_witness_two, &mut fq_sponge, domain, &srs);
+
+        let folding_compat_constraints: Vec<FoldingCompatibleExpr<MIPSFoldingConfig>> = constraints
+            .iter()
+            .map(|x| FoldingCompatibleExpr::from(x.clone()))
+            .collect::<Vec<_>>();
+
+        let (folding_scheme, _) = FoldingScheme::<MIPSFoldingConfig>::new(
```
Comment (querolita) [inline o1vm/src/mips/tests.rs:520]: I suppose the next step will be to have the same example but using the decomposable scheme instead, right? I am curious to see if it works for ADDIU or not.

File: o1vm/src/mips/tests.rs [hunk 2]
```diff
@@ -323,4 +335,200 @@ mod folding {
...
+        });
+        // sanity check
+        witness
+            .cols
+            .iter()
+            .for_each(|x| assert_eq!(x.len(), domain_size));
+        witness
+    }
+
+    fn build_folding_instance(
```
Comment (querolita) [inline o1vm/src/mips/tests.rs:390]: Don't you prefer to implement `Foldable` for the normal `Trace`? I can adapt the signature so it matches better what you have in here.

**Review by querolita — APPROVED** [review]

File: o1vm/src/mips/tests.rs (hunk 1 above)
Comment (dannywillems) [reply o1vm/src/mips/tests.rs:519]: Yes. Which would be duplicated what you did in Keccak

File: o1vm/src/mips/tests.rs (hunk 2 above)
Comment (dannywillems) [reply o1vm/src/mips/tests.rs:389]: It would be nice to merge, yes. I was thinking we could do a better interface in the near future.

**Comment by dannywillems** [comment]
Back to solving conflicts

**Comment by dannywillems** [comment]
I hope it is fine now!!!

**Pull request merged**